repairnator/kubernetes-support/README.md

```diff
@@ -0,0 +1,171 @@
+```
+	kubectl get pods
+	kubectl exec -it mongo-XXXXXXX-XXXXX bash
+	mongo
+```
+Then we need to create a database called "githook" (note that it must be named like this according to the script if you don't plan to change it) and insert some information into a collection called "gitSecrets". These are done by these steps below.
+```
+	use githook
+	db.gitSecrets.insert({"slugName" : "Spirals-Team/repairnator", "gitSecret" : "123456789"})
+```
+Next is deploying the repo monitor itself on K8s. First you need to go to the "repomonitor-dockerimage" to build and push an image to docker just like how we did with the scanner. Then there is also a yaml file in "repairnator-deployment-yamlfiles" folder for deploying and also here we need to replace in the "repo-monitor.yaml" file the name of your image you tagged before when pushing to docker (format DOCKER_USER_NAME/IMAGE_NAME). Then apply the yaml 
+```
+	kubectl create -f repo-monitor.yaml
+```
+This will create a pod named "repo-monitor-XXXXXXXX-XXXX" together with a service creating a Loadbalancer with an external address for github. We need to fetch this address by this command 
+```
+	kubectl get svc 
+```
+and look for a service named "repo-monitor" then take its external IP (it usually takes a while before the LoadBalancer finish setting up).
+Then go to your repo github page -> setting -> webhook -> add new webhook and then add the Payload URL" as "http://LoadBalancer_IP/" shown in the picture below, "Content type" is set as "application/json" then secret is the "gitSecret" previously added to the database "123456789" (Note: this should be a hash of some kind like a MD5 hash to be safer) then also select "Let me select individual events" to scroll down a bit to select "Statuses". The reason we choose only sending status update is because the server is set to only activate the pipeline when the build status is failure.
+![repo-monitor-picture](repo-monitor-example.png)
+
+Now back to the terminal. We can check the logs of our repo-monitor deployed earlier by first 
+```
+	kubectl logs -f repo-monitor-XXXXXXXX-XXXX
+```
+You should be able to see something like
+```
+Listening on port 30050!
+Authorized
+Sending scan request for slug Spirals-Team/repairnator
+```
+Same logic apply for any repo want to be monitored. First register the slug in database with a secret then add a webhook like mentioned before then we are done. 
+
+## Setup Mongodb (optional)
```

Comment:
I would suggest to provide two different ways to setup the PV&PVC: using Google cloud service, or using a simple nfs system. I will send an example yaml file to you later.

Reply:
The example could be found [here](https://github.com/gluckzhang/repairnator/commit/21cd702730ae30b0d4f56879bd1889e32a93a356).

Reply:
Ah ok. I do not have knowledge about this nfs thing. Do you know some simple example with this so I can play around and familiarize myself with this ? :). 

PS: I'll add this as an alternative way to set this up as you suggested :) .